packages/synapse-react-client/src/lib/containers/oauth/CreateOAuthClient.tsx

```diff
@@ -53,6 +57,31 @@ export const CreateOAuthModal: React.FunctionComponent<
   const warningBody =
     'Editing this detail will render your client invalid and will require you to resubmit verification. This action cannot be undone.'
   const uriHelpMessage = 'Click Add URI to add more Redirect URIs'
+
+  // Return the OAuth Client definition based on the current client-side UI state
+  const getOAuthClient = useCallback(() => {
+    const oAuthClient: OAuthClient = {
+      client_id: client?.client_id,
+      client_name: clientName,
+      redirect_uris: redirectUris?.map(str => str.uri) ?? [''],
+      policy_uri: policyUri,
+      client_uri: clientUri,
+      sector_identifier_uri: sectorUri ?? '',
+      tos_uri: tosUri,
+      etag: client?.etag,
+    }
+    return oAuthClient
+  }, [
+    client?.client_id,
+    client?.etag,
+    clientName,
+    clientUri,
+    policyUri,
+    redirectUris,
+    sectorUri,
+    tosUri,
+  ])
```

Comment:
This is being invoked every render, so can simplify to `useMemo`
```suggestion
  const oAuthClient: OAuthClient = useMemo(() => {
    return {
      client_id: client?.client_id,
      client_name: clientName,
      redirect_uris: redirectUris?.map(str => str.uri) ?? [''],
      policy_uri: policyUri,
      client_uri: clientUri,
      sector_identifier_uri: sectorUri ?? '',
      tos_uri: tosUri,
      etag: client?.etag,
    }
  }, [
    client?.client_id,
    client?.etag,
    clientName,
    clientUri,
    policyUri,
    redirectUris,
    sectorUri,
    tosUri,
  ])
```